bin/brew

```diff
@@ -6,16 +6,19 @@ chdir() {
 BREW_FILE_DIRECTORY="$(chdir "${0%/*}" && pwd -P)"
 HOMEBREW_BREW_FILE="$BREW_FILE_DIRECTORY/${0##*/}"
 
+HOMEBREW_PREFIX="${HOMEBREW_BREW_FILE%/*/*}"
+
 if [[ -L "$HOMEBREW_BREW_FILE" ]]
 then
   BREW_SYMLINK="$(readlink "$HOMEBREW_BREW_FILE")"
   BREW_SYMLINK_DIRECTORY="$(dirname "$BREW_SYMLINK")"
   BREW_FILE_DIRECTORY="$(chdir "$BREW_FILE_DIRECTORY" &&
                          chdir "$BREW_SYMLINK_DIRECTORY" && pwd -P)"
+  HOMEBREW_REPOSITORY="${BREW_FILE_DIRECTORY%/*}"
+else
+  HOMEBREW_REPOSITORY="${HOMEBREW_PREFIX}"
```

Comment:
I was thinking this could be moved directly underneath the `HOMEBREW_PREFIX=` line above the `if` to emphasize that this is the regular case. The `if` would then be dedicated to handle the less common case of `brew` being a symlink, which would include updating the contents of `HOMEBREW_REPOSITORY`.

Also, purely stylistic but the curly braces can be omitted to be more in line with the existing code.